viziphant/gpfa.py

```diff
@@ -956,46 +984,66 @@ def _set_axis_labels_trajectories(ax,
-                                gpfa_instance):
-
-    trial_events = block_with_cut_trials.segments[trial_idx].filter(
-        objects='Event',
-        name=neo_event_name)[0]
-
-    # get mask for the relevant events
-    mask = np.zeros(trial_events.array_annotations['trial_event_labels'].shape,
-                    dtype='bool')
-    for event in relevant_events:
-        mask = np.logical_or(
-            mask,
-            trial_events.array_annotations['trial_event_labels'] == event)
+                                neo_event_dict,
+                                gpfa_instance,
+                                verbose):
+    
+    
+    trial = block_with_cut_trials.segments[trial_idx]
+
+    event_times = []
+    event_labels = []
+    for event_label, neo_event_properties in neo_event_dict.items():
+        events = neo.utils.get_events(trial, **neo_event_properties)
+        if len(events) == 0:
+            if verbose:
+                print(f'No event found for label {event_label}.')
+            continue
+        if len(events[0].times) > 1 and verbose:
+            print(f'More than one events for label {event_label}.',
+                   'Proceed by choosing the first one.')
+        event_times.append(events[0].times[0])
+        event_labels.append(event_label)
 
     # cheating by converting event times to binned spiketrain
-    t_start = block_with_cut_trials.segments[trial_idx].t_start
-    t_stop = block_with_cut_trials.segments[trial_idx].t_stop
-
-    event_spiketrain = neo.SpikeTrain(trial_events.times[mask],
-                                      t_start=t_start,
-                                      t_stop=t_stop)
+    event_spiketrain = neo.SpikeTrain(event_times,
+                                      units=trial.spiketrains[0].units,
+                                      t_start=trial.t_start,
+                                      t_stop=trial.t_stop)
     bin_size = gpfa_instance.bin_size
     binned_event_spiketrain = BinnedSpikeTrain(
         event_spiketrain,
         bin_size=bin_size).to_array().flatten()
 
     time_bins_with_relevant_event = np.nonzero(binned_event_spiketrain)[0]
-    relevant_event_labels = \
-        trial_events.array_annotations['trial_event_labels'][[mask]]
+    # relevant_event_labels = \
+    #     trial_events.array_annotations['trial_event_labels'][[mask]]
 
-    return time_bins_with_relevant_event, relevant_event_labels
+    return time_bins_with_relevant_event, event_labels
 
 
-def _show_unique_legend(axes):
+def _show_unique_legend(axes, legend_args=dict()):
     # only plot unique labels
     handles, labels = axes.get_legend_handles_labels()
     if len(handles) == 0:
         # no labels have been provided
         return
     by_label = dict(zip(labels, handles))
-    axes.legend(by_label.values(), by_label.keys())
+    
+    axes.legend(by_label.values(), by_label.keys(), **legend_args)
+
+def determine_trials_to_plot(trials_to_plot, n_trials):
```

Comment:
```suggestion
def _determine_trials_to_plot(trials_to_plot, n_trials):
```

make helper function private ?